app/cdap/components/DataPrep/Directives/ChangeDataType/DecimalOptions.tsx

```diff
@@ -0,0 +1,293 @@
+            arrow
+            placement="top"
+            disableFocusListener={!isRoundingDisabled}
+            disableHoverListener={!isRoundingDisabled}
+            disableTouchListener={!isRoundingDisabled}
+            title={T.translate(`${PREFIX}.roundingDisabledTooltip`).toString()}
+            classes={{ tooltip: classes.customTooltip, arrow: classes.customTooltipArrow }}
+          >
+            <FormControl variant="outlined" disabled={isRoundingDisabled}>
+              <InputLabel id={ROUNDING_LABEL_ID} htmlFor={ROUNDING_INPUT_ID}>
+                {T.translate(`${PREFIX}.roundingLabel`)}
+              </InputLabel>
+              <Select
+                labelId={ROUNDING_LABEL_ID}
+                id={ROUNDING_INPUT_ID}
+                value={rounding}
+                onChange={handleRoundingChange}
+                label={T.translate(`${PREFIX}.roundingLabel`)}
+                endAdornment={
+                  <InputAdornment position="end">
+                    <Tooltip
+                      arrow
+                      enterDelay={500}
+                      title={T.translate(`${PREFIX}.roundingTooltip`).toString()}
+                      classes={{
+                        tooltip: classes.customTooltip,
+                        arrow: classes.customTooltipArrow,
+                      }}
+                    >
+                      <IconButton
+                        aria-label={T.translate(`${PREFIX}.roundingHelpButton`).toString()}
+                        edge="end"
+                      >
+                        <HelpIcon />
+                      </IconButton>
+                    </Tooltip>
+                  </InputAdornment>
+                }
+              >
+                {ROUNDING_MODES.map((mode: IRoundingMode) => (
+                  <MenuItem value={mode.value} key={mode.value}>
+                    <Tooltip
+                      arrow
+                      enterDelay={500}
+                      title={mode.description}
+                      classes={{
+                        tooltip: classes.customTooltip,
+                        arrow: classes.customTooltipArrow,
+                      }}
+                    >
+                      <span>{mode.text}</span>
+                    </Tooltip>
+                  </MenuItem>
+                ))}
+              </Select>
+              <FormHelperText>{T.translate(`${PREFIX}.roundingHelperText`)}</FormHelperText>
+            </FormControl>
+          </Tooltip>
+          <ButtonsContainer>
+            <Button variant="contained" color="primary" disableElevation onClick={applyDirective}>
+              {T.translate(`${PREFIX}.applyButton`)}
+            </Button>
+            <Button color="primary" onClick={onCancel}>
+              {T.translate(`${PREFIX}.cancelButton`)}
+            </Button>
+          </ButtonsContainer>
+        </FormContainer>
+      </Paper>
+    </SubmenuContainer>
+  );
+};
+
+const FormContainer = styled.div`
```

Comment:
can we pull these into a separate file?

Reply:
sure. Makes sense to keep the styled components in a separate file.